### PR TITLE
Eventengine/state simplification

### PR DESCRIPTION
**Pull request by MikeDobrzan**
Simplify calls and syntax

### CODE REVIEW

File: src/Api/PubnubApi/EventEngine/Subscribe/Events/SubscriptionEvents.cs [hunk 1]
```diff
@@ -4,7 +4,6 @@ namespace PubnubApi.PubnubEventEngine.Subscribe.Events {
 	public class SubscriptionChangedEvent : Core.IEvent {
 		public IEnumerable<string> Channels;
 		public IEnumerable<string> ChannelGroups;
-		public SubscriptionCursor Cursor;
```
Comment (budgetpreneur) [inline src/Api/PubnubApi/EventEngine/Subscribe/Events/SubscriptionEvents.cs:7]: Cursor is removed?

File: src/Api/PubnubApi/EventEngine/Subscribe/Invocations/SubscriptionInvocations.cs [hunk 2]
```diff
@@ -3,17 +3,38 @@
...
+		{
+			this.Messages = messages;
+		}
 	}
 
 	internal class EmitStatusInvocation : Core.IEffectInvocation {
-		public IEnumerable<string> Channels;
-		public IEnumerable<string> ChannelGroups;
+		// TODO merge status variables into one?
+		public PNStatusCategory StatusCategory;
```
Comment (budgetpreneur) [inline src/Api/PubnubApi/EventEngine/Subscribe/Invocations/SubscriptionInvocations.cs:16]: Do we need separate PNStatusCategory when we already have Category in PNStatus?

File: src/Api/PubnubApi/EventEngine/Subscribe/States/ReceiveFailedState.cs [hunk 3]
```diff
@@ -3,66 +3,44 @@
...
+        public Tuple<Core.IState, IEnumerable<IEffectInvocation>> Transition(IEvent e)
+        {
+            return e switch
+            {
+                Events.SubscriptionChangedEvent subscriptionChanged => new ReceivingState()
+                {
+                    Channels = subscriptionChanged.Channels,
+                    ChannelGroups = subscriptionChanged.ChannelGroups,
+                    Cursor = this.Cursor
+                }.With(null),
```
Comment (budgetpreneur) [inline src/Api/PubnubApi/EventEngine/Subscribe/States/ReceiveFailedState.cs:26]: not sure if you want to clean up With(null)

File: src/Api/PubnubApi/EventEngine/Subscribe/States/ReceiveFailedState.cs [hunk 4]
```diff
@@ -3,66 +3,44 @@
...
-			}
-		}
-	}
-}
+                Events.ReconnectEvent reconnect => new ReceivingState()
+                {
+                    Channels = reconnect.Channels,
+                    ChannelGroups = reconnect.ChannelGroups,
+                    Cursor = reconnect.Cursor
+                }.With(null),
```
Comment (budgetpreneur) [inline src/Api/PubnubApi/EventEngine/Subscribe/States/ReceiveFailedState.cs:33]: not sure if you want to clean up With(null)

File: src/Api/PubnubApi/EventEngine/Subscribe/States/ReceiveFailedState.cs [hunk 5]
```diff
@@ -3,66 +3,44 @@
...
 
+                Events.SubscriptionRestoredEvent subscriptionRestored => new ReceivingState()
+                {
+                    Channels = subscriptionRestored.Channels,
+                    ChannelGroups = subscriptionRestored.ChannelGroups,
+                    Cursor = subscriptionRestored.Cursor
+                }.With(null),
```
Comment (budgetpreneur) [inline src/Api/PubnubApi/EventEngine/Subscribe/States/ReceiveFailedState.cs:40]: not sure if you want to clean up With(null)

File: src/Api/PubnubApi/EventEngine/Subscribe/Events/SubscriptionEvents.cs (hunk 1 above)
Comment (MikeDobrzan) [reply src/Api/PubnubApi/EventEngine/Subscribe/Events/SubscriptionEvents.cs:7]: Yes, according to the docs. At least "for now".

File: src/Api/PubnubApi/EventEngine/Subscribe/Invocations/SubscriptionInvocations.cs (hunk 2 above)
Comment (MikeDobrzan) [reply src/Api/PubnubApi/EventEngine/Subscribe/Invocations/SubscriptionInvocations.cs:16]: This is still used by disconnect/reconnect, will see if we can further simplify

**Review by budgetpreneur — APPROVED** [review]
LGTM

**Pull request merged**